addons/web_editor/static/src/js/backend/form_controller.js

```diff
@@ -0,0 +1,32 @@
+odoo.define('web_editor.FormController', function (require) {
+"use strict";
+const FormController = require('web.FormController');
+
+FormController.include({
+
+    /**
+     * @override
+     */
+    saveRecord: function () {
+        return this._super(...arguments).then((changedFields) => {
+            // update res_id with the actual record id for the Attachment, when record going to save
+            // this will required for showing attachment under the media dialog
+            // because of the attachment already created with res_id = 0
+            const res_id = this.model.get(this.handle, {raw: true}).res_id;
+            const wysiwygAttachmentsID = this.renderer.wysiwygAttachmentsID || [];
+            if (wysiwygAttachmentsID.length && res_id){
+                return this._rpc({
+                    model: 'ir.attachment',
+                    method: 'write',
+                    args: [Array.from(wysiwygAttachmentsID, (attachment) => attachment.id), {
+                        res_id: res_id,
+                    }],
```

Comment:
This should definitely not be done in JS, as I explained the attachment ids must be sent to the python with the rest of the "save" query (whatever that is in this case), and ensure the attachments can actually be written this way.

Here either the user won't have access right to attachment and this will fail, or he will have access to them and then he would be able to overwrite the res_id of attachment already linked to other records by mistake (or not).